### PR TITLE
ci(action): add deleted/renamed url in comment

### LINKED ISSUE #463: Simplify doc content reviewer life: warn about aliases when pages have changed
In the past, we saw that aliases are often missing. People are not aware that they must configure redirects for moved or deleted pages.
As we did with #373, we should remind/warn contributors when aliases are required.

In the same PR comment we created with #373, we could include a dedicated paragraph about the pages that need an alias.
My proposal is to list the url of former pages, the reviewer then click on the link to check that the redirect is setup (remember that Antora generates redirects that also work in the Surge preview) and goes to the right destination page.

In the future, we could automate that the redirect is correctly setup by running http get commands to at least detect missing alias. But reviewers will always as to check the destination because we cannot automate the check (always specific to the use case)

### PR DESCRIPTION
**Pull request by benjaminParisel**
Improve the action that manages the PR comment about the details of the preview which is deployed to Surge:
* List deleted/renamed URL (on GitHub, a renamed file is detected as deleted and created with a new name)
* URL could be used to check if alias is exist and functional
* Only notify changes about AsciiDoc files (don't consider images for instance)

closes #463

### CODE REVIEW

**Comment by benjaminParisel** [comment]
A test PR on Bonita-doc component is ready [here](https://github.com/bonitasoft/bonita-doc/pull/2248)

File: .github/actions/comment-pr-with-links/comments-with-url-links.js [hunk 1]
```diff
@@ -1,51 +1,56 @@
...
         'In order to merge this pull request, you need to check your updates with the following url.\n\n';
 
-    const availableLinks = `### :mag: Page list: \n ${links}\n\n\n\n`;
-
+    const availableLinks = `### :mag: Page list: \n ${links.updated}\n\n\n`;
     let warningAliasMessage = '';
-    if(hasWarningMessage){
-        warningAliasMessage='\n \n ### :warning: At least one page has been deleted in the Pull Request. Make sure to add [aliases](https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc#use-alias-to-create-redirects)'
+    if (hasWarningMessage) {
+        warningAliasMessage = `### :warning: Alias \n At least one page has been renamed, moved or deleted in the Pull Request. Make sure to add [aliases](https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc#use-alias-to-create-redirects) and verify if this following link redirect in the right place \n ${links?.deleted}`
```
Comment (tbouffard) [inline .github/actions/comment-pr-with-links/comments-with-url-links.js:39]: I suggest we introduce redirects in the title as aliases are just the way to create the redirects. We mention 'aliases' in the description

```suggestion
        warningAliasMessage = `### :warning: Check redirects \n At least one page has been renamed, moved or deleted in the Pull Request. Make sure to add [aliases](https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc#use-alias-to-create-redirects) and verify that the following links redirect to the right location: \n ${links?.deleted}`
```

File: .github/actions/comment-pr-with-links/comments-with-url-links.js [hunk 2]
```diff
@@ -1,51 +1,56 @@
...
 };
 
-function buildMessage(header,links,hasWarningMessage){
+function buildMessage({header, links, hasWarningMessage}) {
     const preface =
         'In order to merge this pull request, you need to check your updates with the following url.\n\n';
 
-    const availableLinks = `### :mag: Page list: \n ${links}\n\n\n\n`;
-
+    const availableLinks = `### :mag: Page list: \n ${links.updated}\n\n\n`;
```
Comment (tbouffard) [inline .github/actions/comment-pr-with-links/comments-with-url-links.js:36]: I suggest we provide more context here, as we do for redirects/aliases.

```suggestion
    const availableLinks = `### :mag: Updated pages \n The following pages were updated, please ensure that the display is correct: \n ${links.updated}\n\n\n`;
```

File: .github/actions/comment-pr-with-links/comments-with-url-links.js [hunk 3]
```diff
@@ -1,51 +1,60 @@
...
-
+    const availableLinks = `### :mag: Updated pages 
+    The following pages were updated, please ensure that the display is correct: 
+    ${links.updated}\n\n`;
     let warningAliasMessage = '';
-    if(hasWarningMessage){
-        warningAliasMessage='\n \n ### :warning: At least one page has been deleted in the Pull Request. Make sure to add [aliases](https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc#use-alias-to-create-redirects)'
+    if (hasWarningMessage) {
+        warningAliasMessage = `### :warning: Check redirects
+         At least one page has been renamed, moved or deleted in the Pull Request. Make sure to add [aliases](https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc#use-alias-to-create-redirects) and verify hat the following links redirect to the right location:
```
Comment (tbouffard) [inline .github/actions/comment-pr-with-links/comments-with-url-links.js:42]:
```suggestion
         At least one page has been renamed, moved or deleted in the Pull Request. Make sure to add [aliases](https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc#use-alias-to-create-redirects) and verify that the following links redirect to the right location: 
```

File: .github/actions/comment-pr-with-links/comments-with-url-links.js [hunk 4]
```diff
@@ -1,51 +1,60 @@
...
-function buildMessage(header,links,hasWarningMessage){
+function buildMessage({header, links, hasWarningMessage}) {
     const preface =
         'In order to merge this pull request, you need to check your updates with the following url.\n\n';
 
-    const availableLinks = `### :mag: Page list: \n ${links}\n\n\n\n`;
-
+    const availableLinks = `### :mag: Updated pages 
+    The following pages were updated, please ensure that the display is correct: 
+    ${links.updated}\n\n`;
```
Comment (tbouffard) [inline .github/actions/comment-pr-with-links/comments-with-url-links.js:38]:
```suggestion
    ${links.updated}
`;
```

File: .github/actions/comment-pr-with-links/comments-with-url-links.js [hunk 5]
```diff
@@ -1,51 +1,60 @@
...
     let warningAliasMessage = '';
-    if(hasWarningMessage){
-        warningAliasMessage='\n \n ### :warning: At least one page has been deleted in the Pull Request. Make sure to add [aliases](https://github.com/bonitasoft/bonita-documentation-site/blob/master/docs/content/CONTRIBUTING.adoc#use-alias-to-create-redirects)'
+    if (hasWarningMessage) {
+        warningAliasMessage = `### :warning: Check redirects
```
Comment (tbouffard) [inline .github/actions/comment-pr-with-links/comments-with-url-links.js:41]:
```suggestion
        warningAliasMessage = `
         ### :warning: Check redirects
```

**Comment by benjaminParisel** [comment]
@tbouffard i have added your suggestion directly in my last comment.

I update the pattern to filter only *.adoc files, see the following before my updated

image

**Comment by tbouffard** [comment]
I guess we must remove the indentation in the "template string" :-)

image

**Comment by benjaminParisel** [comment]
image

I think it's better :smile:

**Review by tbouffard — APPROVED** [review]
Very nice work ⭐

**Pull request merged**